src/main/java/org/datadog/jenkins/plugins/datadog/DatadogMetricCounter.java

```diff
@@ -0,0 +1,43 @@
+package org.datadog.jenkins.plugins.datadog;
+
+import hudson.Extension;
+import hudson.model.*;
+import org.datadog.jenkins.plugins.datadog.clients.DatadogHttpClient;
+import java.util.concurrent.ConcurrentMap;
+import java.util.logging.Logger;
+import org.datadog.jenkins.plugins.datadog.model.LocalCacheCounters;
+
+@Extension
+public class DatadogMetricCounter extends PeriodicWork {
```

Comment:
I am still not convinced about the name of this class. It submits counters. Maybe we should call this `PeriodicCounterForwarder` or something similar. Also, in order to be more generic we could change the logic a little bit so that we also specify the metric name for each count.